src/screens/Signin.js

```diff
@@ -9,35 +9,85 @@ import React from 'react';
 import { t } from 'ttag'
 
 import logo from '../assets/images/hathor-logo.png';
+import hathorLib from '@hathor/wallet-lib';
 import InitialImages from '../components/InitialImages';
+import { setInitWalletName } from '../actions/index';
+import { connect } from "react-redux";
 
 
+const mapDispatchToProps = dispatch => {
+  return {
+    setInitWalletName: name => dispatch(setInitWalletName(name)),
+  };
+};
+
 /**
  * Screen used to select between create a new wallet or import an old one
  *
  * @memberof Screens
  */
 class Signin extends React.Component {
+  /**
+   * errorMessage {string} Message to be shown when an error happens
+   */
+  state = { errorMessage: '' };
+
+  /**
+   * Check wallet name is not blank or already in use
+   */
+  validateWalletName = (name) => {
```

Comment:
Should we have a validation for forbidden characters?

Also, do we have any words we would like to reserve?
Once we deliver we will not be able to cleanly forbid words since users may have them already added.

I saw that we use `maxLength` on the input, but should we have this validation here so that we can control the error message and format?